tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -0,0 +1,268 @@
+                 Tommy Nguyen Azure Automation SE
+                 Atchut Barli Azure Automation Sr. SE
+                 Shipra Malik Ohri Azure Automation Sr. SE
+                 Shujun Liu Azure Automation Sr. SE
+                 Adrian Doyle Sr. Esc. Eng.
+                 Brian McDermott Sr. Esc. Eng.
+
+   Date        : 2018-07-06
+   Version     : 1.0
+   
+'''
+
+import sys
+import os
+import os.path
+import socket
+import datetime
+
+from os import walk
+
+def main(output_path=None):
+    output = []
+
+    if os.geteuid() != 0:
+        print "Please run this script in sudo"
+        exit()
+
+    output.append(get_machine_info() + "\n")
+    output.append(check_oms_admin() + "\n")
+    output.append(check_oms_agent_installed() + "\n")
+    output.append(check_oms_agent_running() + "\n")
+    output.append(check_hybrid_worker_installed() + "\n")
+    output.append(check_hybrid_worker_running() + "\n")
+    output.append(check_network() + "\n")
+
+    output.append("\n")
+
+    if output_path is not None:
+        try: 
+            os.makedirs(output_path)
+        except OSError:
+            if not os.path.isdir(output_path):
+                raise
+        f = open(output_path + "/healthcheck-" + str(datetime.datetime.utcnow().isoformat()) + ".log", "w")
+        f.write("".join(output))
+
+    print "".join(output)
+
+def check_endpoints(workspace, endpoints, success_message, failure_message):
+    output = []
+
+    for endpoint in endpoints:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        new_endpoint = None
+        
+        if "*" in endpoint and workspace is not None:
+            new_endpoint = endpoint.replace("*", workspace)
+        elif "*" not in endpoint:
+            new_endpoint = endpoint
+
+        if new_endpoint is not None:
+            try:
+                response = sock.connect_ex((new_endpoint, 443))
+
+                if response == 0:
+                    output.append(new_endpoint + ": " + success_message + "\n")
+                else:
+                    output.append(new_endpoint + ": " + failure_message + "\n")
+
+            except Exception as ex:
+                output.append(new_endpoint + ": " + failure_message + "\n")
+                output.append(str(ex) + "\n")
+
+    return output
+
+def check_network():
+    output = []
+    output.append("Network check: \n\n")
+
+    endpoints = ["bing.com", "google.com"]
+    agent_endpoints = []
+    jrds_endpoints = []
+
+    agent_endpoint = get_agent_endpoint()
+    jrds_endpoint = get_jrds_endpoint()
+
+    if agent_endpoint is not None:
+        agent_endpoints.append(agent_endpoint)
+    
+    if jrds_endpoint is not None:
+        jrds_endpoints.append(jrds_endpoint)
+
+    ods_endpoints = ["*.ods.opinsights.azure.com", "*.oms.opinsights.azure.com", "ods.systemcenteradvisor.com"]
+
+    ff_endpoints = ["usge-jobruntimedata-prod-1.usgovtrafficmanager.net", "usge-agentservice-prod-1.usgovtrafficmanager.net", 
```

Comment:
can we just confirm with  a security expert on exposing us gov endpoints in public file?

Reply:
per email with Matt Hofacker, he linked this https://docs.microsoft.com/en-us/azure/azure-government/documentation-government-services-monitoringandmanagement . since these seem to have the FF endpoints, I think we are good.